python/mxnet/gluon/rnn/rnn_layer.py

```diff
@@ -186,11 +186,13 @@ def forward(self, inputs, states=None):
                 self.i2h_weight[i].shape = (self._gates*self._hidden_size, inputs.shape[2])
                 self.i2h_weight[i]._finish_deferred_init()
         if inputs.context.device_type == 'gpu' or \
-           self._mode == 'lstm' and not (self._dropout and autograd.is_training()):
+           self._mode in ['lstm', 'gru'] and \
+           not (self._dropout and autograd.is_training()):
```

Comment:
I just realized that unlike LSTM, GRU inference doesn't have dropout support yet, so this condition needs fixing.